relay-server/src/actors/envelopes.rs

```diff
@@ -937,11 +951,18 @@ impl EnvelopeProcessor {
     }
 
     /// Remove profiling items if the feature flag is not enabled
+    #[cfg(feature = "processing")]
     fn process_profiling_items(&self, state: &mut ProcessEnvelopeState) {
         let profiling_enabled = state.project_state.has_feature(Feature::Profiling);
         state.envelope.retain_items(|item| {
             match item.ty() {
                 ItemType::ProfilingSession | ItemType::ProfilingTrace => profiling_enabled,
+                ItemType::Profile => {
+                    if !profiling_enabled {
+                        return false;
+                    }
+                    matches!(self.parse_profile(item), Ok(_))
```

Comment:
FYI: An alternative way of writing this is below, both are fine.

```suggestion
                    self.parse_profile(item).is_ok()
```